common/service/sensor/sensor.h

```diff
@@ -230,19 +231,31 @@ typedef struct _sensor_cfg__ {
 	bool is_enable_polling;
 	int cache;
 	uint8_t cache_status;
-	bool (*pre_sensor_read_hook)(uint8_t, void *);
+	bool (*pre_sensor_read_hook)(void *, void *);
 	void *pre_sensor_read_args;
-	bool (*post_sensor_read_hook)(uint8_t, void *, int *);
+	bool (*post_sensor_read_hook)(void *, void *, int *);
 	void *post_sensor_read_args;
 	void *init_args;
 
 	/* if there is new parameter should be added, please add on above */
 	void *priv_data;
 	uint8_t retry;
 	uint8_t (*init)(uint8_t, int *);
-	uint8_t (*read)(uint8_t, int *);
+	uint8_t (*read)(void *, int *);
```

Comment:
These read functions should take a non-`void *` to me.

Reply:
Hi @williamspatrick ,

Because we need to input the cfg pointer parameter in the read function to adapt for various sensor monitor tables, and the read function is a parameter in the sensor cfg structure, so we can only replace the parameters to be put  into the read function with void pointer.

Regards,
Lora.

Reply:
Can you give some concrete examples?  What are the different types you need to pass in to various functions?

Reply:
Oh, I think you’re saying that it is self referencing?  In that case we should just break it into two structs. One with just the properties and one with the property-struct plus methods.

Reply:
> Oh, I think you’re saying that it is self referencing? In that case we should just break it into two structs. One with just the properties and one with the property-struct plus methods.

Yes !
Would you like to add another similar cfg structure to input the structure including how to read the sensors ?